### DISCUSSION TITLE
启动 H:\work\git\ant>bin\msvc\debug\lua.exe test/simple/main.lua 出错

### DISCUSSION THREAD

**Discussion by freeman53** [ejoy/ant#50] [Q&A]
stack traceback:
        ( service:6 )
        /engine/task/service/service.lua:821: in for iterator 'for iterator'
        H:/work/git/ant/pkg/ant.compile_resource/parallel_task.lua:14: in field 'wait'
        ...rk/git/ant/pkg/ant.compile_resource/material/convert.lua:21: in field '?'
        H:/work/git/ant/pkg/ant.compile_resource/main.lua:55: in field 'compile_file'
        engine/game/io.lua:166: in upvalue 'COMPILE'
        engine/game/io.lua:62: in upvalue 'getresource'
        engine/game/io.lua:82: in upvalue 'getfile'
        engine/game/io.lua:91: in local 's'
        ( service:11 )
        [string "package.path = "/engine/?.lua"..."]:7: in upvalue 'call'
        [string "package.path = "/engine/?.lua"..."]:15: in field 'read'
        H:/work/git/ant/pkg/ant.io/main.lua:7: in upvalue 'readall'
        H:/work/git/ant/pkg/ant.io/main.lua:12: in field 'readall'
        ...ork/git/ant/pkg/ant.resource_manager/thread/material.lua:223: in upvalue 'material_create'
        ...ork/git/ant/pkg/ant.resource_manager/thread/material.lua:248: in local 's'
        ( service:9 )
        H:/work/git/ant/pkg/ant.asset/async.lua:13: in field 'load_material'
        H:/work/git/ant/pkg/ant.imgui/imgui_system.lua:27: in local 'f'
        H:/work/git/ant/pkg/ant.ecs/main.lua:389: in function <H:/work/git/ant/pkg/ant.ecs/main.lua:384>
        H:/work/git/ant/pkg/ant.ecs/main.lua:573: in upvalue 'system_changed'
        H:/work/git/ant/pkg/ant.ecs/main.lua:579: in method 'pipeline_init'
        H:/work/git/ant/pkg/ant.window/service/world.lua:80: in upvalue 'f

**Comment by cloudwu** [comment]
编译资源出错多半是没有构建工具：

```
luamake -mode release tools
```
